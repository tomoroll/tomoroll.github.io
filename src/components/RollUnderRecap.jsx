import React from 'react';
import PropTypes from 'prop-types';
import { getProfit } from '../utils/etheroll-contract';


function RollUnderRecap({ betSize, value }) {
  const chances = value - 1;
  const profit = getProfit(betSize, chances);
  return (
    <div className="row">
      <div className="col-6">
        <h3>Roll under</h3>
      </div>
      <div className="col-6">
        <h3 className="text-right">{value}</h3>
      </div>
      <div className="col-6">
        With a wager of
      </div>
      <div className="col-6">
        <p className="text-right mb-0">
          {betSize.toFixed(8)}
          &nbsp;
          ETH
        </p>
      </div>
      <div className="col-6">
        For a profit of
      </div>
      <div className="col-6">
        <p className="text-right">
          {profit.toFixed(8)}
          &nbsp;
          ETH
        </p>
      </div>
    </div>
  );
}
RollUnderRecap.propTypes = {
  betSize: PropTypes.number.isRequired,
  value: PropTypes.number.isRequired,
};

export default RollUnderRecap;
