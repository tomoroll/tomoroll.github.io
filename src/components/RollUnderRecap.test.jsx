import React from 'react';
import renderer from 'react-test-renderer';
import RollUnderRecap from './RollUnderRecap';

it('renders correctly', () => {
  const tree = renderer
    .create(<RollUnderRecap value={51} betSize={0.1} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
