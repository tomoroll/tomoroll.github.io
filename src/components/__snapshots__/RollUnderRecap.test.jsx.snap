// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-6"
  >
    <h3>
      Roll under
    </h3>
  </div>
  <div
    className="col-6"
  >
    <h3
      className="text-right"
    >
      51
    </h3>
  </div>
  <div
    className="col-6"
  >
    With a wager of
  </div>
  <div
    className="col-6"
  >
    <p
      className="text-right mb-0"
    >
      0.10
        ETH
    </p>
  </div>
  <div
    className="col-6"
  >
    For a profit of
  </div>
  <div
    className="col-6"
  >
    <p
      className="text-right"
    >
      0.10
        ETH
    </p>
  </div>
</div>
`;
